generic.go

```diff
@@ -0,0 +1,83 @@
+package kindsys
+
+import (
+	"bytes"
+	"encoding/gob"
+
+	jsoniter "github.com/json-iterator/go"
+)
+
+// Verify this implements a real resource
+var _ Resource = &GenericResource[any, CustomMetadata, any]{}
+
+// GenericResource
+type GenericResource[Spec any, CustomMeta CustomMetadata, Status any] struct {
+	StaticMeta StaticMetadata
+	CommonMeta CommonMetadata
+	CustomMeta CustomMeta
+	Spec       Spec
+	Status     Status
+}
+
+func (u *GenericResource[Spec, CustomMeta, Status]) SpecObject() any {
+	return u.Spec
+}
+
+func (u *GenericResource[Spec, CustomMeta, Status]) Subresources() map[string]any {
+	return map[string]any{
```

Comment:
Do we expect/want `GenericResource` to handle other non-status subresources as well in a `any` kind of capacity? Or is the idea here that this only represents known, typed data?

Reply:
TBD for sure, but in general I think we should be very careful to add additional properties (sub-resources) that are expected to be saved on the resource and are not compatible with standard CRD storage